node-support/src/effect-serializer.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const AnySupport = require("./protobuf-any");
+const util = require("util");
+
+module.exports = class EffectSerializer {
+
+  constructor(allEntities) {
+    this.allEntities = allEntities;
+  }
+
+  serializeEffect(method, message) {
+    let serviceName, commandName;
+    // We support either the grpc method, or a protobufjs method being passed
+    if (typeof method.path === "string") {
+      const r = new RegExp("^/([^/]+)/([^/]+)$").exec(method.path);
```

Comment:
Are `RegExp`s not reusable?

Reply:
Yeah it should be pulled out into a constant, but not for performance reasons, v8 caches compiled regexes:

```javascript
> console.time("1"); new RegExp("^/([^/]+)/([^/]+)$").exec("foo/bar"); console.timeEnd("1");
1: 0.108ms
undefined
> console.time("1"); new RegExp("^/([^/]+)/([^/]+)$").exec("foo/bar"); console.timeEnd("1");
1: 0.011ms
undefined
```

Reply:
Interesting!